src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -219,15 +224,17 @@ def generate_update_stmt(
 
     def all_value_map(self, row: Row) -> Dict[str, Any]:
         """
-        Takes a row and preserves only the fields that are defined in the Dataset
-        and are not flagged as read-only. Used for scenarios when an update endpoint
-        has required fields other than just the fields being updated.
+        Takes a row and preserves only the fields that are defined in the Dataset.
+        Used for scenarios when an update endpoint has required fields other than
+        just the fields being updated.
         """
         all_value_map: Dict[str, Any] = {}
         for field_path, field in self.field_map().items():
-            # only map scalar fields that are not read-only
-            if isinstance(field, ScalarField) and not field.read_only:
-                # only map if the value exists on the row
+            # only map scalar fields
+            if (
+                isinstance(field, ScalarField)
+                and pydash.get(row, field_path.string_path) is not None
+            ):
```

Comment:
maybe i'm missing something, but isn't this check just repeated on the follow line? i can understand grouping them together in the same conditional, but i don't think we gain anything by repeating it, and the code becomes a bit less intuitive.

Reply:
i think GH was playing a joke on me - the UI glitched and had me add this comment twice. of course it was a comment that was pointing out redundancy in the code :D